packaging/flavorgen/flavors/generators.go

```diff
@@ -104,6 +103,9 @@ var (
 		regexVar(vSphereTemplateVar),
 		regexVar(vSphereHaproxyTemplateVar),
 	}
+	k8sVersionToCPIVersion = map[string]string{
+		"v1.18.6":"gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.2.1",
```

Comment:
as you see in #1059 we're moving away from baking the CPI into CAPV

@andrewsykim @dvonthenen - would it make sense to move/re-tag the CPI with minor kubernetes version ? if not, at worst consumers may maintain that mapping

Reply:
> would it make sense to move/re-tag the CPI with minor kubernetes version ? 

Yes, that is what this PR is trying to do. Going forward we are always tracking the Kubernetes minor versions, so `v1.18.6` needs to map to v1.18.0 (when released). However, the patch versions may not match so we still need to map a given Kubernetes minor version with the latest CPI patch version. See [this KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cloud-provider/1771-versioning-policy-for-external-cloud-providers) for more details. 

> if not, at worst consumers may maintain that mapping

I'm not sure we want to put this burden on users to be honest. I think the CRS approach is reasonable, as long as we do the versioning mapping there. @yastij would it be reasonable for @gwang550 to have a follow-up PR to update the generated CRS code when yours is merged? When are we expecting that to merge and are we going to remove the baked in deployment right after? If we think the baked in deployment is staying for a while it's valuable to make the necessary changes here, if we are replacing it entirely with CRS, let's have a follow-up PR for the CRS changes.

Reply:
#1045 outlines the timeline for CPI and CSI

@andrewsykim - I expect the PR to merge today, I just need to wrap up a few things. Note that consumers that don't generate templates programmatically through something similar to https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/tree/master/packaging/flavorgen might not benefit from @gwang550's PR

Reply:
Not this specific PR, but we need to add logic in the template generate to pick the right version

Reply:
@yastij Let me if there is something that I need to pick up following your PR. In the meantime, I am going to look into how to corporate the logic into your PR